target/linux/realtek/files-6.6/drivers/i2c/busses/i2c-gpio-shared.c

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Bitbanging driver for multiple I2C busses with shared SCL pin using the GPIO API
+ * Copyright (c) 2025 Markus Stockhausen <markus.stockhausen at gmx.de>
+ */
+
+#include <linux/i2c-algo-bit.h>
+#include <linux/gpio/consumer.h>
+#include <linux/mod_devicetable.h>
+#include <linux/mutex.h>
+#include <linux/platform_device.h>
+
+#define SI2C_MAX_BUS	4
+
+struct si2c_ctx;
```

Comment:
If you start with struct `si2c_ctx`, you don't have to forward declare it.

Reply:
This is needed because the ctx/bus link each other. Changing order will not help.